src/R/Core/Impl/AST/Operators/Associativity.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.R.Core.AST.Operators {
+    // https://en.wikipedia.org/wiki/Operator_associativity
+    /// <summary>
+    /// Associativity. Consider the expression a~b~c. If the operator 
+    /// ~ has left associativity, this expression would be interpreted as (a~b)~c.
+    /// If the operator has right associativity, the expression would be
+    /// interpreted as a~(b~c).
+    /// </summary>
+    public enum Associativity {
```

Comment:
It appears that R also has non-associative operators. Try:
```r
1 == 2 == TRUE
```
and it fails, with a parser error at that (can be verified by using `quote` to make sure that no evaluation takes place).